### DISCUSSION TITLE
I've succeeded with compiling the same hello world bionic binary on- and off- device

### DISCUSSION THREAD

**Discussion by t184256** [nix-community/nix-on-droid#186] [Show and tell]
I've succeeded with compiling the same hello world bionic binary on- and off- device
https://github.com/t184256/nix-compile-for-android @ 034be52
That paves the road to a proot that's buildable/overrideable/etc on device if need be.

A bit of a sad news is that it takes ~300 MB downloads, ~2 GB disk space and quite some time.

**Comment by Gerschtli** [comment]
Awesome! That means we don't have to hardcode proot's store path anymore, right? As long as we have populated the binary cache, I don't see a problem with the build time.

**Comment by t184256** [comment]
So far I haven't fit inside my mind what it means because building it two different ways causes a content hash match, not a nix store hash match. Thus is sounds like it simplifies all on-device affairs, bootstrap archive building can become more complicated.

So either need to keep a fixed-hash junction mechanism to allow building from either arch and try to avoid hardcoding the hash somehow; or we build it on emulated aarch64, maybe.

Yeah, and the cache will have to be kept up to date with nixpkgs advancing or be pinned.

Or we can have a switch between fixed and own-built proot.

As I said, I haven't figured it all out yet.
